add generic browser query to manifest (fix #12209)

moving-bits opened the pull request:
## Description
For testing whether the "open in browser not working" is related to not having listed those apps in our manifest, this PR adds
- Firefox browser
- Samsung browser
- Via browser

to the `<query>` section of our manifest to make those packages visible to c:geo

@Lineflyer 
Can you cross-check whether this works in your test scenario or in those of the two users on support?

Lineflyer commented:
Tested and seems to work.
On the prepared emulator I installed VIA and "Duck Duck Go browser". 
Running `release` it says "No browser found", running this PR it looks as expected:

![image](https://user-images.githubusercontent.com/949669/144724930-32a336d4-50ca-4090-a2c7-d6d4f16861ae.png)

Lineflyer commented:
Might this help: https://developer.android.com/training/package-visibility/use-cases#check-browser-available

fm-sys commented:
> Might this help: https://developer.android.com/training/package-visibility/use-cases#check-browser-available

This looks like a perfect solution, thanks for finding it 👍



> The list doesn't include browser apps if the user has configured the URL to open in a non-browser app by default.

Just for info: We query for browser apps via https://example.com, so we will be able to find browser apps, even if cgeo is set as default app.

moving-bits commented:
Updated the PR to implement the generic query for browsers as described above, removed the explicitly listed extra browsers, and updated the PR title accordingly.

Lineflyer commented:
Tested on my emulator setup and works:
Now the browser selection always shows whatever browser I have selected as default on the device.

So on Android 11 you will now always get two options:
- c:geo internal    
- The default browser

That's IMHO sufficient for users.
